extensions/charts/ranchercisbenchmark.go

```diff
@@ -0,0 +1,215 @@
+	ClusterScanReportType      = "cis.cattle.io.clusterscanreport"
+)
+
+// ClusterScanStatus represents the status field of a cluster scan object.
+type ClusterScanStatus struct {
+	Display Display
+}
+
+// Display contains the state of the cluster scan.
+// State can be pending, running, reporting, pass, and fail
+type Display struct {
+	State   string
+	Message string
+}
+
+// ClusterScanReportSpec represents the specification for a cluster scan report.
+type ClusterScanReportSpec struct {
+	ReportJson string
+}
+
+// CisReport is the report structure stored as report json in cluster scan report spec.
+type CisReport struct {
+	Total         int
+	Pass          int
+	Fail          int
+	Skip          int
+	Warn          int
+	NotApplicable int
+	Results       []*Group `json:"results"`
+}
+
+// Group is the result structure stored as report json in Results of CisReport
+type Group struct {
+	ID     string      `yaml:"id" json:"id"`
+	Text   string      `json:"description"`
+	Checks []*CisCheck `json:"checks"`
+}
+
+// CisCheck is the ID, Description and State structure of individual test in cluster scan.
+type CisCheck struct {
+	Id          string
+	Description string
+	State       string
+}
+
+// InstallRancherCisBenchmarkChart is a helper function that installs the rancher-CIS Benchmark chart.
+func InstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
+	serverSetting, err := client.Management.Setting.ByID(serverURLSettingID)
+	if err != nil {
+		return err
+	}
+
+	registrySetting, err := client.Management.Setting.ByID(defaultRegistrySettingID)
+	if err != nil {
+		return err
+	}
+
+	cisbenchmarkChartInstallActionPayload := &payloadOpts{
+		InstallOptions:  *installOptions,
+		Name:            RancherCisBenchmarkName,
+		Namespace:       RancherCisBenchmarkNamespace,
+		Host:            serverSetting.Value,
+		DefaultRegistry: registrySetting.Value,
+	}
+
+	chartInstallAction := newCisBenchmarkChartInstallAction(cisbenchmarkChartInstallActionPayload)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	// Install the Rancher CIS Benchmark chart
+	err = catalogClient.InstallChart(chartInstallAction, catalog.RancherChartRepo)
+	if err != nil {
+		return err
+	}
+
+	// Wait for chart to be fully deployed
+	watchAppInterface, err := catalogClient.Apps(RancherCisBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherCisBenchmarkName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusDeployed) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Register cleanup function for uninstallation
```

Comment:
also is not necessary. A better comment here would be why the cleanup function isn't set elsewhere, like before chart install. in other words, why does it need to be explicitly set here vs. at the beginning of the function? i.e., what if one of the above functions returns an error. Does that mean no part of the chart was installed, and thus no cleanup is necessary? Sometimes when testing charts I've seen the crd chart installed, but the main chart fail.